api/python/provisioner/commands/create_user.py

```diff
@@ -0,0 +1,365 @@
+
+        local_minion = local_minion_id()
+        cluster_path = PillarKey('cluster')
+        get_result = PillarResolver(local_minion).get([cluster_path])
+        cluster_pillar = get_result[local_minion][cluster_path]
+        nodes_list = [
+            node for node in cluster_pillar.keys()
+            if 'srvnode-' in node
+        ]
+
+        if (not cluster_pillar or
+                cluster_pillar is values.MISSED):
+            raise BadPillarDataError(
+                'value for {} is not specified'.format(cluster_path.pi_key)
+            )
+
+        def _prepare_user_fileroots_dir():
+            StateFunExecuter.execute(
+                'file.directory',
+                fun_kwargs=dict(
+                    name=str(user_fileroots_dir),
+                    makedirs=True
+                )
+            )
+
+        def _generate_ssh_keys():
+            StateFunExecuter.execute(
+                'cmd.run',
+                fun_kwargs=dict(
+                    name=(
+                        f"ssh-keygen -f {keyfile} "
+                        "-q -C '' -N '' "
+                        "-t rsa -b 4096 <<< y"
+                    )
+                )
+            )
+            StateFunExecuter.execute(
+                'ssh_auth.present',
+                fun_kwargs=dict(
+                    # name param is mandetory and expects ssh key
+                    # but ssh key is passed as source file hence name=None
+                    name=None,
+                    user=uname,
+                    source=str(keyfile_pub),
+                    config=str(user_fileroots_dir / 'authorized_keys')
+                )
+            )
+
+        def _generate_ssh_config():
+            for node in nodes_list:
+                hostname = PillarKey(
+                    'cluster/' + node + '/hostname'
+                )
+
+                hostname_pillar = PillarResolver(LOCAL_MINION).get([hostname])
+                hostname_pillar = next(iter(hostname_pillar.values()))
+
+                if (not hostname_pillar[hostname] or
+                        hostname_pillar[hostname] is values.MISSED):
+                    raise BadPillarDataError(
+                        'value for {} is not specified'.format(hostname.pi_key)
+                    )
+
+                ssh_config = f'''Host {node} {hostname_pillar[hostname]}
+    Hostname {hostname_pillar[hostname]}
+    User {uname}
+    UserKnownHostsFile /dev/null
+    StrictHostKeyChecking no
+    IdentityFile {ssh_dir}/{keyfile.name}
+    IdentitiesOnly yes
+    LogLevel ERROR
+    BatchMode yes'''
+
+                StateFunExecuter.execute(
+                    'file.append',
+                    fun_kwargs=dict(
+                        name=str(user_fileroots_dir / 'config'),
+                        text=ssh_config
+                    )
+                )
+
+        def _copy_minion_nodes():
+            StateFunExecuter.execute(
```

Comment:
What are we copying here. Add a docstring or comment.

Reply:
Addressed